dbms/src/Common/SharedLibrary.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <boost/core/noncopyable.hpp>
+
+namespace DB
+{
+/** Allows you to open a dynamic library and get a pointer to a function from it.
+  */
+class SharedLibrary : private boost::noncopyable
+{
+public:
+    SharedLibrary(const std::string & path);
+
+    ~SharedLibrary();
+
+    void * getImpl(const std::string & name, bool no_throw);
+
+    template <typename Func>
+    Func get(const std::string & name, bool no_throw = false)
```

Comment:
Pair of methods `get` and `tryGet` is better than `no_throw` param.
(leave `no_throw` param only for impl-method)